vividus-plugin-azure-storage-account/src/main/java/org/vividus/azure/storage/blob/BlobStorageSteps.java

```diff
@@ -231,6 +238,77 @@ public void findBlobs(StringComparisonRule rule, String blobNameToMatch, String
+     *
+     * @param filter            Filter to apply to blob names.
+     *                          <div>Example:</div>
+     *                          <code>
+     *                          <br>When I filter blobs by:
+     *                          <br>|blobNamePrefix|blobNameFilterRule|blobNameFilterValue|resultsLimit|
+     *                          <br>|data/         |contains          |file-key.txt       |10          |
+     *                          <br>in container `global` of storage account `storage`
+     *                          <br> and save result to story variable `blobs`
+     *                          </code>
+     *                          <br>
+     *                          <br>where all filters are optional, but at least one rule is required.
+     *                          <ul>
+     *                          <li><code>blobNameFilterRule</code> The blob name comparison rule: "is equal to",
+     *                           "contains", "does not contain" or "matches".
+     *                           Should be specified along with <i>blobNameFilterValue</i>.</li>
+     *                          <li><code>blobNameFilterValue</code> The full or partial blob name to be matched.
+     *                           Should be specified along with <i>blobNameFilterRule</i>.</li>
+     *                          <li><code>blobNamePrefix</code> The prefix which blob names should start with.</li>
+     *                          <li><code>resultsLimit</code> Maximum number of blob names to return.</li>
+     *                          </ul>
+     * @param containerName     The name of the container to point to.
+     * @param storageAccountKey The key to Storage Account endpoint.
+     * @param scopes            The set (comma separated list of scopes e.g.: STORY, NEXT_BATCHES) of the variable
+     *                          scopes.<br>
+     *                          <i>Available scopes:</i>
+     *                          <ul>
+     *                          <li><b>STEP</b> - the variable will be available only within the step,
+     *                          <li><b>SCENARIO</b> - the variable will be available only within the scenario,
+     *                          <li><b>STORY</b> - the variable will be available within the whole story,
+     *                          <li><b>NEXT_BATCHES</b> - the variable will be available starting from next batch
+     *                          </ul>
+     * @param variableName      The variable name to store the list of found blob names.
+     */
+    @When("I filter blobs by:$filter in container `$containerName` of storage account `$storageAccountKey`"
+            + " and save result to $scopes variable `$variableName`")
+    public void findBlobs(BlobFilter filter, String containerName, String storageAccountKey, Set<VariableScope> scopes,
+            String variableName)
+    {
+        filter.validate();
```

Comment:
How do you think, maybe we could introduce the converter, it will simplify a few things:
1. Validation will be performed at the converter level
2. We could combine name filter and rule into POJO, because they work only in pairs, or we could even create the  Optional<Matcher> nameFilter right in the converter. This approach will simplify this https://github.com/vividus-framework/vividus/pull/1972/files#diff-514588a30d1bc0c47c22777fb464ba7548bd428d25bd80277f4ca8cd0981dd34R298-R302